apps/explorer/src/components/ownedobjects/views/DynamicFieldsCard.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { useRef, useCallback, useEffect } from 'react';
+
+import { UnderlyingObjectCard } from './UnderlyingObjectCard';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import { DisclosureBox } from '~/ui/DisclosureBox';
+import { ObjectLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+
+const MAX_PAGE_SIZE = 20;
+
+export function DynamicFieldsCard({ id }: { id: string }) {
+    const rpc = useRpcClient();
+    const {
+        data,
+        isInitialLoading,
+        isFetchingNextPage,
+        hasNextPage,
+        fetchNextPage,
+    } = useInfiniteQuery(
+        ['dynamic-fields', id],
+        ({ pageParam = null }) =>
+            rpc.getDynamicFields({
+                parentId: id,
+                cursor: pageParam,
+                limit: MAX_PAGE_SIZE,
+            }),
+        {
+            enabled: !!id,
+            getNextPageParam: ({ nextCursor, hasNextPage }) =>
+                hasNextPage ? nextCursor : null,
+        }
+    );
+
+    const observerElem = useRef<HTMLDivElement | null>(null);
+
+    const handleObserver = useCallback(
+        (entries: IntersectionObserverEntry[]) => {
+            const [target] = entries;
+            if (target.isIntersecting && hasNextPage) {
```

Comment:
do we need to check here also that it doesn't already fetch then next page? something like `!isFetchingNextPage`

Reply:
I think we need to check that we are not already loading the nextPage otherwise it will trigger multiple requests



https://user-images.githubusercontent.com/10210143/231284004-f1893c38-2851-48a0-a7d0-e9dd286cc6fd.mov